# Wet Lab Resources for Molecular Biologists

*Last updated August 2023.*

Here are a list of online resources that should help you find information about DNA part sequences, molecular biology questions or basic protocols for handling DNA/RNA/proteins and _E. coli_, and to help you to begin to troubleshoot your own experiments. Most resources are freely available. Note that this list isn't meant to be exhaustive, it is mainly to get you started.

------------------------------------------------------------------------

### General Reference and Tools

-   Bionumbers - <https://bionumbers.hms.harvard.edu/search.aspx>
-   Calculators - conversions <http://www.genscript.com/conversion.html>, <https://labhacks.net/#calculators>
-   PCR calculator - <https://ec363.shinyapps.io/pcr_calc/>
-   Lab Hacks app - <https://apps.apple.com/us/app/lab-hacks/id1462593060#?platform=iphone>

------------------------------------------------------------------------

### MolBio Reference and Tools

**Codes and abbreviations**

-   Nucleotide abbreviations - <https://en.wikipedia.org/wiki/Nucleotide#Abbreviation_codes_for_degenerate_bases>
    -   DNA modifications (IDT) <https://eu.idtdna.com/site/Catalog/Modifications/>
-   Modified dNTPs (and similar) - <https://www.jenabioscience.com/nucleotides-nucleosides/nucleotides-by-structure>
-   Amino acids (abbreviations and properties) - <https://en.wikipedia.org/wiki/Proteinogenic_amino_acid#Chemical_properties>
-   Codon Table - <https://en.wikipedia.org/wiki/DNA_codon_table>

**Codon usage tables**

-   _E. coli_ and others
    -   K-12 <http://www.kazusa.or.jp/codon/cgi-bin/showcodon.cgi?species=83333>
    -   B <http://www.kazusa.or.jp/codon/cgi-bin/showcodon.cgi?species=413997>
    -   W3310 <http://www.kazusa.or.jp/codon/cgi-bin/showcodon.cgi?species=316407>
    -   all _E. coli_ <http://www.kazusa.or.jp/codon/cgi-bin/spsearch.cgi?species=escherichia+coli&c=i>
    -   K-12 <http://openwetware.org/wiki/Escherichia_coli/Codon_usage>
    -   unspecified strain of _E. coli_: <https://www.genscript.com/tools/codon-frequency-table>
-   Rare codon analysis tool- <https://www.genscript.com/tools/rare-codon-analysis>

**Tools**

-   Clustal Omega, DNA/RNA/Protein alignment <http://www.ebi.ac.uk/Tools/msa/clustalo/>
-   Blast, sequence search in databases <http://blast.ncbi.nlm.nih.gov/Blast.cgi>
-   Back-translation <http://www.ebi.ac.uk/Tools/st/>

------------------------------------------------------------------------

### General Lab Techniques

**Buffers**

-   Buffers for biochemical reactions -
    -   <https://www.promega.co.uk/resources/product-guides-and-selectors/protocols-and-applications-guide/buffers-for-biochemical-reactions/>
    -   <http://www.sigmaaldrich.com/life-science/core-bioreagents/biological-buffers/learning-center/buffer-reference-center.html>
    -   <http://www.sigmaaldrich.com/life-science/core-bioreagents/biological-buffers/learning-center/buffer-calculator.html>

**Protocol databases**

-   Addgene protocols (molbio and cloning) - <https://www.addgene.org/protocols/>
-   Addgene guides - <https://www.addgene.org/guides/>
-   Addgene ebooks - <https://www.addgene.org/educational-resources/ebooks/>
-   Addgene Plasmids 101 blog - <https://blog.addgene.org/topic/plasmids-101>
-   Protocols.io - <https://www.protocols.io/>
-   CSHL Protocols (mainly molbio) - <http://cshprotocols.cshlp.org/site/misc/subject.xhtml>
-   OpenWetWare - <http://openwetware.org/wiki/Protocols>
-   Barrick Lab protocols (microbiology) - <https://barricklab.org/twiki/bin/view/Lab/ProtocolList>

**Protocol FAQs**

-   ResearchGate Forum (Sign up to let you browse Q&As better) - <https://www.researchgate.net/topics>

------------------------------------------------------------------------

### Molecular Biology Techniques

-   Introduction to recombinant DNA technology by Addgene <https://www.addgene.org/mol-bio-reference/>
-   Enzymes used in molecular biology: a useful guide <https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2570007/>
-   Why Johnny can't clone: Common pitfalls and not so common solutions <https://www.future-science.com/doi/10.2144/000114324>

------------------------------------------------------------------------

### CLONING - DNA Construct Design

**Software:**

-   Benchling (free) - Labbook and DNA analysis package. <https://www.benchling.com/>
-   Snapgene (£££) - DNA analysis package. <http://www.snapgene.com/support/tutorial_videos/introduction_to_snapgene/>

**Plasmids:**

Plasmid databases:

-   commercial
    -   <https://www.neb.com/tools-and-resources/interactive-tools/dna-sequences-and-maps-tool>
    -   <https://www.embl.de/pepcore/pepcore_services/strains_vectors/vectors/bacterial_expression_vectors/popup_bacterial_expression_vectors/>
-   donated <https://www.addgene.org/browse/>
-   other <https://dnasu.org/DNASU/Home.do> , <https://dnasu.org/DNASU/SearchOptions.do?tab=1>

Plasmid parts: The vector

-   What is a plasmid <https://blog.addgene.org/plasmids-101-what-is-a-plasmid>
-   Origin of replication <http://blog.addgene.org/plasmid-101-origin-of-replication>
-   Antibiotic resistance gene - <https://blog.addgene.org/plasmids-101-everything-you-need-to-know-about-antibiotic-resistance-genes>
    -   Antibiotic working stocks - <https://www.addgene.org/mol-bio-reference/antibiotics/>
    -   though Cam is often used at 34µg/ml final, slightly more than 25 µg/ml.

Plasmid parts: The construct

-   **Databases and lists of common parts:**
-   iGEM Registry databases and distributions: <http://parts.igem.org/assembly/libraries.cgi>.
-   Sequences of common parts
    -   Snapgene -\> Detect common features
    -   Addgene's list <https://www.addgene.org/tools/reference/plasmid-features/>
-   Promoters <https://blog.addgene.org/plasmids-101-the-promoter-region>
    -   Andersen promoters (set of constititive promoters from weak to strong): iGEM pages.
-   RBS design calculators
    -   <https://www.denovodna.com/software/>
    -   <https://salislab.net/software/>
    -   <http://wolfson.huji.ac.il/expression/vector/RibosomalBindingSites.html>
-   Common Epitope Tags
    -   <https://www.addgene.org/mol-bio-reference/#tags>

**Cloning methods:**

-   Types of cloning
    -   <https://www.addgene.org/mol-bio-reference/cloning/>
    -   <https://www.snapgene.com/features/#beyond-the-basics-of-molecular-cloning>
-   Traditional cloning (Type II restriction digest)
    -   <https://www.addgene.org/mol-bio-reference/cloning/#re>
    -   <https://www.addgene.org/protocols/subcloning/>
    -   <https://www.addgene.org/protocols/pcr-cloning/>
-   **Type IIS cloning combined with PCR cloning (= recommended as default)**
    -   <https://www.addgene.org/mol-bio-reference/cloning/#typeIIs>
-   Oligo cloning
    -   <https://www.addgene.org/protocols/annealed-oligo-cloning/>
-   Gibson assembly (also recommended)
    -   <https://www.addgene.org/protocols/gibson-assembly/>
    -   <http://openwetware.org/wiki/Janet_B._Matsen:Guide_to_Gibson_Assembly>
-   SPLiCE (by Edo Gianni) - <http://openwetware.org/wiki/SPLiCE>

**Primer design:**

-   General rules on primer design (but keep Tm differences to 3oC max)- <https://www.addgene.org/protocols/primer-design/>
-   Primer analysis
    -   Snapgene has Tm calculations that work OK.
    -   IDT is more thorough: <https://eu.idtdna.com/calc/analyzer>
    -   with formulae: <http://www.genscript.com/cgi-bin/tools/primer_calculation>
-   Primer design for annealed oligo cloning <https://www.addgene.org/protocols/annealed-oligo-cloning/>

**DNA sequence optimisation tools:**

-   Synonymise- small R tool I wrote to randomise codons where repetitive codon usage was not desired- <https://github.com/ec363/synonymise>
-   DNA Chisel- Python tool for avoiding homologies, tuning GC content, codon optimisation- <https://github.com/Edinburgh-Genome-Foundry/DnaChisel>

**_E. coli_ strains:**

-   _E. coli_ strains for cloning: <https://www.addgene.org/mol-bio-reference/#strains>
-   NEB 10beta- <https://www.neb.com/products/c3020-neb-10-beta-electrocompetent-e-coli>
-   NEB T7 express lysY/lacIq- <https://www.neb.com/products/c3013-t7-express-lysyiq-competent-e-coli-high-efficiency>
-   BL21(DE3)- <https://www.neb.com/products/c2527-bl21de3-competent-e-coli>
-   More strains: <https://www.addgene.org/mol-bio-reference/strain-information/>
-   Common lab strains: <https://blog.addgene.org/plasmids-101-common-lab-e-coli-strains>

------------------------------------------------------------------------

### CLONING - In Practice

**General cloning protocols:**

Essential basic protocols:

-   Overview <https://www.addgene.org/protocols/>
-   General Cloning Protocols:
-   **Use these for basic principles only. Always follow reagent-specific instructions for reaction mixture recipes and thermocycler protocols**
-   PCR <https://www.addgene.org/protocols/pcr/>
-   Restriction digests <https://www.addgene.org/protocols/restriction-digest/>
-   DNA analysis by agarose gel electrophoresis <https://www.addgene.org/protocols/gel-electrophoresis/>
-   DNA ligation <https://www.addgene.org/protocols/dna-ligation/>
-   Bacterial transformation using electroporation: **see wet-lab-protocols**
-   Bacterial transformation using heat shock <https://www.addgene.org/protocols/bacterial-transformation/>
-   Overnight bacterial cultures <https://www.addgene.org/protocols/inoculate-bacterial-culture/>
-   Creating glycerol stocks <https://www.addgene.org/protocols/create-glycerol-stock/>
-   Streaking a plate from a glycerol stock <https://www.addgene.org/protocols/streak-plate/>
-   Minipreps: DNA purification from bacterial culture <https://www.addgene.org/protocols/purify-plasmid-dna/>
-   Quantification of DNA using a Nanodrop/spectrophotometer <https://www.addgene.org/protocols/dna-quantification/>
-   Sequence analysis of DNA <https://www.addgene.org/protocols/sequence-analysis/>
-   More:
-   Phenol-chloroform extraction and ethanol precipitation <https://www.addgene.org/protocols/purify-plasmid-dna/>

Reagent and kit-specific information:

-   Reagent-specific information can be found on the manufacturer's website.
-   Kits: Minipreps, PCR purification, Gel extraction - All include manuals within the boxes, which can also be found online.

More information:

-   Typical DNA marker ladders:
    -   1kb ladder- <https://www.neb.com/products/n3232-1-kb-dna-ladder>
    -   low MW ladder- <https://www.neb.com/products/n3233-low-molecular-weight-dna-ladder>
-   Restriction enzymes:
    -   SnapGene, EnzymeX, NEB website all have good info on these
    -   <https://www.neb.com/tools-and-resources/usage-guidelines/nebuffer-performance-chart-with-restriction-enzymes>
-   Specific PCR protocols
    -   Q5 - <https://www.neb.com/protocols/2012/08/30/pcr-using-q5-hot-start-high-fidelity-dna-polymerase-m0493>
    -   Taq - <https://international.neb.com/protocols/0001/01/01/taq-dna-polymerase-with-standard-taq-buffer-m0273>
    -   PCR calculator - <https://ec363.shinyapps.io/pcr_calc/>
-   Sequencing
    -   Follow instructions on provider's webpage about how to prepare samples. They often have stock primers for standard plasmids.
-   Protocols for automating DNA assembly with OpenTrons: <https://github.com/BASIC-DNA-ASSEMBLY/DNA-BOT>

------------------------------------------------------------------------

### MICROBIOLOGY

-   Commandments of experimental microbiology (Barrick) <https://barricklab.org/twiki/bin/view/Lab/StandardMicrobiologicalPractices>
-   OD600 calculator <http://www.genomics.agilent.com/biocalculators/calcODBacterial.jsp>

**_E. coli_ strains/genotypes:**

-   List of strain genotypes <https://openwetware.org/wiki/E._coli_genotypes>
-   What the genotypes mean <http://openwetware.org/wiki/Escherichia_coli/Nomenclature_%26_Abbreviations>
-   List of _E. coli_ strains, but no links: <http://www.uniprot.org/taxonomy/83333>

**_E. coli_ genome/genes:**

-   EcoGene (when it was live it was great) <http://www.ecogene.org/>

**_E. coli_ general:**

-   openwetware (protocols): <http://openwetware.org/wiki/Escherichia_coli>
-   wiki: <http://ecoliwiki.net/colipedia/index.php/Welcome_to_EcoliWiki>
    -   strains <http://ecoliwiki.net/colipedia/index.php/Category:Strains>
    -   gene lists <http://ecoliwiki.net/colipedia/index.php/Category:Gene_Lists>
    -   genomes: <http://ecoliwiki.net/colipedia/index.php/Category:E._coli_genomes>
    -   DH10b
        -   <http://ecoliwiki.net/colipedia/index.php/DH10B>
        -   genome <https://www.ncbi.nlm.nih.gov/nucleotide/NC_010473>
        -   ref <http://ecoliwiki.net/colipedia/index.php/PMID:18245285>
    -   BL21 (DE3)
        -   <https://www.ncbi.nlm.nih.gov/nuccore/NC_012971>
-   EcoCyc: <http://ecocyc.org/>
    -   literature-based curation of the K12-MG1655 genome
-   Characteristics of the _E. coli_ genome (Barrick) <https://barricklab.org/twiki/bin/view/Lab/ReferenceEColiGenome>

------------------------------------------------------------------------

### PROTEIN EXPRESSION (_E. coli_)

Optimisation (maximisation):

-   Genscript's description is an overview of some considerations for maximisation & includes both mammalian and bacterial elements <https://www.genscript.com/codon-optimization-for-increased-protein-expression.html>

Protein electrophoresis:

-   typical protein ladder: <https://www.neb.com/products/p7712-color-prestained-protein-standard-broad-range-11-245-kda>
-   hall of shame: <http://www.ruf.rice.edu/~bioslabs/studies/sds-page/sdsgoofs.html>
-   Tricine-SDS-PAGE for small proteins and peptides \<30 kDa (especially if \<15 kDa, as 15-30 can just about be resolved by standard glycine/Laemmli-SDS-PAGE). This paper is great resource not just for Tricine gels but also for hydrophobic proteins, membrane proteins, SDS, DTT/beta-merc reducing agents, etc. 2006 Schägger et al. Nat Protocols <doi:10.1038/nprot.2006.4>

Troubleshooting:

-   EMBL <https://www.embl.de/pepcore/pepcore_services/protein_expression/ecoli/index.html>
-   RBSCalculators <https://salislab.net/software/>

### PROTEIN PURIFICATION

Cell lysis and clarification:

-   EMBL <https://www.embl.de/pepcore/pepcore_services/protein_purification/extraction_clarification/index.html>
-   EMBL - _E. coli_ lysis w sonication <https://www.embl.de/pepcore/pepcore_services/protein_purification/extraction_clarification/cell_lysates_ecoli/sonication/index.html>
-   EMBL - _E. coli_ lysis w lysozyme <https://www.embl.de/pepcore/pepcore_services/protein_purification/extraction_clarification/cell_lysates_ecoli/enzymatic_lysis/index.html>
-   Wolfson centre protocols <http://wolfson.huji.ac.il/purification/Purification_Protocols.html>

Purification techniques:

-   Common Epitope Tags - <https://www.addgene.org/mol-bio-reference/#tags>
-   Cytiva Handbooks on Principles & Methods in Protein Purification (previously GE Lifesciences/Amersham Biosciences) <https://www.cytivalifesciences.com/en/gb/support/handbooks>
-   EMBL <https://www.embl.de/pepcore/pepcore_services/protein_purification/purification/index.html>

Concentration analysis:

-   Comparison (Wolfson Centre for Applied Structural Biology, Hebrew University of Jerusalem). <http://wolfson.huji.ac.il/purification/Protocols/Comparision_of_Methods.htm>
-   A280 ext coeff: <https://www.expasy.org/resources/protparam>

Storage:

-   EMBL <https://www.embl.de/pepcore/pepcore_services/protein_purification/storage_purified_proteins/index.html>

### PROTEIN ANALYSIS

**Protein structure databases**

-   UniProt <http://www.uniprot.org/>
-   PDB (Europe) <http://www.ebi.ac.uk/pdbe/>
-   RCSB <http://www.rcsb.org/pdb/home/home.do> (Research Collaboratory for Structural Bioinformatics, US)
-   Others (list) - <https://en.wikipedia.org/wiki/Protein_structure_database> of interest:
    -   Proteopedia: essentially a wikipedia for proteins.
    -   ProtCID: Protein common interface database (for homologous proteins).

**Amino acid properties**

-   <http://www.sigmaaldrich.com/life-science/metabolomics/learning-center/amino-acid-reference-chart.html>
-   <https://en.wikipedia.org/wiki/Amino_acid>
-   <https://en.wikipedia.org/wiki/Proteinogenic_amino_acid#Chemical_properties>

**Protein Properties**

-   Protein properties, Expasy <http://www.expasy.org/resources>

**Protein families by structure**

-   SCOP <http://scop2.mrc-lmb.cam.ac.uk/>
-   CATH <http://www.cathdb.info/search>
-   Pfam <http://pfam.xfam.org/>

------------------------------------------------------------------------

### BIOCHEMISTRY/CHEMISTRY

-   Protein-Ligand binding: online simulation of the fraction of protein molecules in ligand-bound state as a function of binding affinity, protein concentration, and ligand concentration <https://binding.streamlit.app>
-   SMILES (chemistry notation) generator <https://www.cheminfo.org/flavor/malaria/Utilities/SMILES_generator___checker/index.html>
-   Small molecule permeability in Gram negative bacteria <http://www.entry-way.org/pages/about>

------------------------------------------------------------------------

### FLUORESCENT PROTEINS

-   FPbase: database of fluorescent proteins <https://www.fpbase.org>

------------------------------------------------------------------------

### EXPERIMENTAL DESIGN

-   Introduction to experimental design (Barrick) <https://barricklab.org/twiki/bin/view/Lab/IntroductionToExperimentalDesign>

------------------------------------------------------------------------

### ANALYSIS

-   R. Recommended over Excel for data analysis and presentation. A guide to downloading R/RStudio and getting started using R is provided on the Coding Resources page: <https://github.com/ec363/coding_resources>
-   GitHub. Again, see the Coding Resources page: <https://github.com/ec363/coding_resources>

-   Paper on Tidy Data (Wickham, 2014) <https://www.jstatsoft.org/article/view/v059i10>
-   **Parsley**: Our web application for parsing data from exported format data from plate readers. Parsing = data extraction + data tidying + metadata joining. App and guide are here: <https://gbstan.shinyapps.io/parsleyapp/> and preprint is here <https://zenodo.org/record/8072500>.

-   **FPCountR**: Our R package for the conversion of fluorescence measurements from microplate reader data from 'arbitrary units' to absolute units (ie. molecules of fluorescent protein), and the analysis of timecourse _E. coli_ data from 'fluorescence/OD' to 'molecules/cell or molar concentration'.
    -   Paper: [Csibra and Stan, 2022. Absolute protein quantification using fluorescence measurements with FPCountR. _Nat Comms_](https://www.nature.com/articles/s41467-022-34232-6)
    -   Code: [FPCountR: FPCountR: Fluorescent protein calibration for plate readers](https://github.com/ec363/fpcountr)
    -   Wet lab protocol: [FPCount protocol - in-lysate (purification free) protocol V.2](https://www.protocols.io/view/fpcount-protocol-in-lysate-purification-free-proto-dm6gpw43plzp/v2)

-   Growth curve analysis with R. <https://github.com/sprouffske/growthcurver>

-   Image analysis with ImageJ <https://imagej.net>

------------------------------------------------------------------------

### FIGURES

-   How to prepare figures (not a bad set of guidelines) <https://barricklab.org/twiki/bin/view/Lab/ProtocolsGraphGuide>

**Photos and Diagrams:**

-   Unsplash (free) - source of free stock photos that work well for presentations. <https://unsplash.com/>
-   Biorender (£££) - scientific illustration application in the browser. expensive but excellent. <https://app.biorender.com/>

**Data visualisation:**

-   [Claus O. Wilke, Fundamentals of Data Visualization](https://clauswilke.com/dataviz/)

